presto-main/src/main/java/com/facebook/presto/execution/scheduler/filter/NodeFilterConfigurationManager.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.execution.scheduler.filter;
+
+import java.util.Optional;
+import java.util.Set;
+
+public interface NodeFilterConfigurationManager
+{
+    /**
+     * Method to get applicable pools given node selection criteria
+     *
+     * @param nodeSelectionCriteria Node selection criteria
+     * @return Applicable pools for the given node selection criteria
+     */
+    Optional<Set<String>> getApplicablePools(NodeSelectionCriteria nodeSelectionCriteria);
```

Comment:
the naming of the interface suggests a way to manager node filter configuration.
The method however suggests doing the actual filtering.
I think there is a mismatch between the naming and the intent.

Reply:
Also getApplicablePools sounds unclear. maybe just say getFilteredPools

Reply:
Based on discussion, I renamed the class to remove filter text from it. These classes may make more sense to be inside scheduler->nodeselection so moved it there. Also it seems like we had used uppercase for the existing package nodeSelection which I refactored to nodeselection as I was adding these classes into that package.